src/borg/archiver/prune_cmd.py

```diff
@@ -163,7 +163,9 @@ def checkpoint_func():
                         log_message = "Keeping archive (rule: {rule} #{num}):".format(
                             rule=kept_because[archive.id][0], num=kept_because[archive.id][1]
                         )
-                if args.output_list:
+                if args.output_list and (
+                    (args.list_pruned and archive in to_delete) or (args.list_kept and archive not in to_delete)
+                ):
```

Comment:
Would it make sense to change the first `and` to `or`?  Then one would just have to supply `--list-pruned` instead of having to supply both `--list-pruned` and `--list`.
```suggestion
                if args.output_list or (
                    (args.list_pruned and archive in to_delete) or (args.list_kept and archive not in to_delete)
                ):
```

Reply:
(The help text would have to change too.)

Reply:
I thought the same, but `output_list` also triggers the setup of a special list-logger elsewhere.

Reply:
That set-up could be done if any of the three options is set.

In any case, I think the code above is not correct.  If only `--list` is provided, it will print nothing, but the docstrings say that both kept and pruned archives will be listed unless one of the other two options is given.

Currently the docs don't match the code, and I don't think either one specifies the most obvious semantics.  How about `--list` means to show both, `--list-kept` means to show the kept ones, and `--list-pruned` means to show the pruned ones.  If more than one is given, they accumulate, rather than subtracting.

Reply:
--list alone works as expected as the default for store_false options is True.
it also works if combined with one of the other options.

but I'll check if i can improve it...

Reply:
uh, wow, something broke tests all over the place...

Reply:
ok, I special-cased these options in implied logging setup now.

Reply:
ok, tests worked. i collapsed the commits into one, guess this is done.

Reply:
Oh, I think I missed that those were `store_false` options.  In any case, I like the new version.